src/Nox.Generator/Domain/EntitiesLocalizedGenerator.cs

```diff
@@ -0,0 +1,49 @@
+﻿using Microsoft.CodeAnalysis;
+using Nox.Generator.Common;
+using Nox.Solution;
+using Nox.Types;
+using System.Linq;
+
+namespace Nox.Generator.Domain;
+
+internal class EntitiesLocalizedGenerator : INoxCodeGenerator
+{
+    public NoxGeneratorKind GeneratorKind => NoxGeneratorKind.Domain;
+
+    public void Generate(SourceProductionContext context, NoxSolutionCodeGeneratorState codeGeneratorState, GeneratorConfig config, string? projectRootPath)
+    {
+        context.CancellationToken.ThrowIfCancellationRequested();
+
+        if (codeGeneratorState.Solution.Domain is null) return;
+
+        foreach (var entity in codeGeneratorState.Solution.Domain.Entities)
+        {
+            // Currently skip owned and composite key entities
+            if (entity.IsOwnedEntity ||
+                entity.Keys.Count > 1)
+            {
+                continue;
+            }
+
+            var entityHasAnyAttributesToLocalize =
```

Comment:
@ILoveMyTortoise  all null checks can be removed, collections must be empty
All the cases you have here should be fixed:

`var entityAttributesToLocalize = entity.AttributesWhere(x => x.Type == NoxType.Text && x.TextTypeOptions.IsLocalized).Select(attributes=>new {Entity = entity, Attributes= attributes});`